### PR TITLE
Improved tests and general docs/ci maintenance

### PR DESCRIPTION
**Pull request by consideRatio**
@manics has made this excellent github action to quickly setup a k8s cluster with k3s, and using that I created a basic test installing both helm charts which validates the k8s resource manifests etc.

Besides that, I try to improve the ease of understanding the CI part of this repo and makes some smaller adjustments like no longer packaging .frigate and README into the helm charts, updating a deprecated resource API version that is safe to upgrade at this point, update badges to GitHub actions, renamed deploy workflow to publish-charts, enabled manual triggering of workflows, etc.

### CODE REVIEW

File: .github/workflows/frigate.yml [hunk 1]
```diff
@@ -1,37 +1,55 @@
...
+#
+name: Ensure charts' README.md are updated
 
 on:
   pull_request:
+    paths:
+      - "*/README.md"
+      - "*/values.yaml"
+      - "*/Chart.yaml"
+  workflow_dispatch:
```
Comment (jacobtomlinson) [inline .github/workflows/frigate.yml:19]: I'm curious why this was added?

File: .github/workflows/frigate.yml [hunk 2]
```diff
@@ -1,37 +1,55 @@
...
+# ref: https://github.com/rapidsai/frigate
+#
+name: Ensure charts' README.md are updated
 
 on:
   pull_request:
+    paths:
+      - "*/README.md"
+      - "*/values.yaml"
+      - "*/Chart.yaml"
```
Comment (jacobtomlinson) [inline .github/workflows/frigate.yml:18]: We should include `*/.frigate` here too in case someone updated the frigate template.

**Review by jacobtomlinson — COMMENTED** [review]
This is an awesome PR, very pleased to see this!

File: .github/workflows/frigate.yml (hunk 1 above)
Comment (consideRatio) [reply .github/workflows/frigate.yml:19]: Ability to press "Run workflow" manually with a button - to have it on that test isnt needed, i just added it as part of adding it to all tests.

Want it removed here?

Comment (consideRatio) [reply .github/workflows/frigate.yml:19]: Removed the manual trigger from this workflow.

File: .github/workflows/frigate.yml (hunk 2 above)
Comment (consideRatio) [reply .github/workflows/frigate.yml:18]: Good catch, added it!

**Comment by consideRatio** [comment]
@jacobtomlinson thanks for the encouraging review, I'm glad you like this PR! :tada: :smiley:

File: .github/workflows/frigate.yml (hunk 1 above)
Comment (jacobtomlinson) [reply .github/workflows/frigate.yml]: > Ability to press "Run workflow" manually with a button

I did not know that was a thing. Nice!

I'm going to merge this now, but if you think it's valuable to add it back in please do so.

**Pull request merged**